query-service-impl/src/test/java/org/hypertrace/core/query/service/trino/QueryRequestToTrinoSQLConverterTest.java

```diff
@@ -0,0 +1,1487 @@
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createFunctionExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createInFilter;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createLongLiteralValueExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createNotEqualsFilter;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createNotInFilter;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createNullNumberLiteralValueExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createNullStringFilter;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createNullStringLiteralValueExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createOrderByExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createStringArrayLiteralValueExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createTimeFilter;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createTimestampFilter;
+import static org.hypertrace.core.query.service.QueryRequestUtil.createContainsKeyFilter;
+import static org.hypertrace.core.query.service.QueryRequestUtil.createNotContainsKeyFilter;
+import static org.hypertrace.core.query.service.QueryRequestUtil.createStringLiteralValueExpression;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.time.Duration;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Optional;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.QueryFunctionConstants;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.Function;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.hypertrace.core.query.service.api.QueryRequest.Builder;
+import org.hypertrace.core.query.service.api.SortOrder;
+import org.hypertrace.core.query.service.trino.converters.TrinoFunctionConverter;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class QueryRequestToTrinoSQLConverterTest {
+
+  private static final String TENANT_ID = "3e761879-c77b-4d8f-a075-62ff28e8fa8a";
+  private static final String TENANT_COLUMN_NAME = "customer_id";
+
+  private static final String TEST_REQUEST_HANDLER_CONFIG_FILE = "trino_request_handler.conf";
+  private static final String TEST_SERVICE_REQUEST_HANDLER_CONFIG_FILE =
```

Comment:
In the follow up, lets keep only `API_TRACE` or `EVENT` level handler in test too. For now, we might not support service handler via trino

Reply:
Added TODO for the same